### PR TITLE
Commerce 12681

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/COMMERCE-12681

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/registry/CommerceProductServiceUpgradeStepRegistrator.java [hunk 1]
```diff
@@ -406,6 +406,10 @@ protected String[][] getTableAndPrimaryKeyColumnNames() {
 			UpgradeProcessFactory.alterColumnName(
 				"CPOption", "DDMFormFieldTypeName",
 				"commerceOptionTypeKey VARCHAR(75) null"));
+		registry.register(
+				"5.10.0", "5.11.0",
+				new com.liferay.commerce.product.internal.upgrade.v5_7_0.
+						CPDefinitionLinkUpgradeProcess(_assetEntryLocalService));
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/registry/CommerceProductServiceUpgradeStepRegistrator.java:412]: This looks wrong, I think you should use the newly created upgrade process instead

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -79,6 +79,14 @@ components:
                     description:
                         URL of the location
                     type: string
+                tags:
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/rest-openapi.yaml:82]: I think this should be `readOnly`

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/AttachmentResourceImpl.java [hunk 3]
```diff
@@ -408,6 +408,13 @@ private Attachment _addOrUpdateAttachment(
 			serviceContext.setExpandoBridgeAttributes(expandoBridgeAttributes);
 		}
 
+		String[] assetTagNames = new String[0];
+
+		if (attachment.getTags() != null) {
+			assetTagNames = attachment.getTags();
+			serviceContext.setAssetTagNames(assetTagNames);
+		}
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/AttachmentResourceImpl.java:416]: Do we really need the variable?

```
if (attachment.getTags() != null) {
	serviceContext.setAssetTagNames(attachment.getTags());
}
```

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/AttachmentResourceImpl.java [hunk 4]
```diff
@@ -442,6 +449,13 @@ private Attachment _addOrUpdateAttachment(
 			serviceContext.setExpandoBridgeAttributes(expandoBridgeAttributes);
 		}
 
+		String[] assetTagNames = new String[0];
+
+		if (attachmentBase64.getTags() != null) {
+			assetTagNames = attachmentBase64.getTags();
+			serviceContext.setAssetTagNames(assetTagNames);
+		}
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/AttachmentResourceImpl.java:457]: Same here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/AttachmentResourceImpl.java [hunk 5]
```diff
@@ -472,6 +486,13 @@ private Attachment _addOrUpdateAttachment(
 			serviceContext.setExpandoBridgeAttributes(expandoBridgeAttributes);
 		}
 
+		String[] assetTagNames = new String[0];
+
+		if (attachmentUrl.getTags() != null) {
+			assetTagNames = attachmentUrl.getTags();
+			serviceContext.setAssetTagNames(assetTagNames);
+		}
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/AttachmentResourceImpl.java:494]: Same here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java [hunk 6]
```diff
@@ -999,6 +999,13 @@ private CPDefinition _updateNestedResources(
 				serviceContext.setExpandoBridgeAttributes(
 					_getExpandoBridgeAttributes(attachment));
 
+				String[] assetTagNames = new String[0];
+
+				if (attachment.getTags() != null) {
+					assetTagNames = attachment.getTags();
+					serviceContext.setAssetTagNames(assetTagNames);
+				}
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:1007]: Same here

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java [hunk 7]
```diff
@@ -1021,6 +1028,13 @@ private CPDefinition _updateNestedResources(
...
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:1036]: Same here

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@andrea-ale-sbarra Please check my comments, thanks!

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Review by alessiorendina — APPROVED** [review]
LGTM! Thank you!

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by pietrobocale** [comment]
ci:test:commerce-functional

**Comment by pietrobocale** [comment]
New PR:
https://github.com/liferay-commerce/liferay-portal/pull/4442

**Comment by pietrobocale** [comment]
ci:close

**Pull request closed without merging**